tango_simlib/model.py

```diff
@@ -307,33 +307,36 @@ class PopulateModelActions(object):
     """Used to populate/update model actions
 
     Populates the model actions using the data from the TANGO device information
-    captured in the POGO generated xmi file.
+    captured in the json file / POGO generated xmi / FANDANGO generated fgo file.
 
     Attributes
     ----------
-    command_info : dict
-        A dictionary of all the device commands together with their
-        metadata specified in the POGO generated XMI file. The key
-        represents the name of the command and the value is a dictionary
-        of all the attribute's metadata.
+    parser_instances : list
+        A list of Parser objects which read an xmi/xml/json file and parses it into device
+        attributes, commands, and properties.
 
     sim_model :  Model instance
         An instance of the Model class which is used for simulation of simple attributes
         and/or commands.
 
     """
-    def __init__(self, parser_instance, tango_device_name, model_instance=None):
-        self.parser_instance = parser_instance
+    def __init__(self, parser_instances, tango_device_name, model_instance=None):
+        self.parser_instances = parser_instances
         if model_instance is None:
             self.sim_model = Model(tango_device_name)
         else:
             self.sim_model = model_instance
         self.add_actions()
 
     def add_actions(self):
-        command_info = self.parser_instance.get_device_command_metadata()
-        override_info = self.parser_instance.get_device_cmd_override_metadata()
+        command_info = {}
+        override_info = {}
+        for parser in self.parser_instances:
+            command_info.update(parser.get_device_command_metadata())
+            override_info.update(parser.get_device_cmd_override_metadata())
```

Comment:
The `self.parser_instances` are only used to get the metadata dicts.  What do you think about passing the dicts directly to `__init__`, rather than the full parser objects?  Of course dicts from all the instances would have to be combined before instantiating the objects.

Reply:
Right. Will do. Same for `PopulateModelProperties`.